### PR TITLE
Document using NodeJS

### PR DESCRIPTION
**Pull request by sharabash**
@ralt please review this and let me know what I should revise or add. Thanks.

### CODE REVIEW

File: reference/nodejs.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Node.js
+
+Node.js is a popular JavaScript runtime built on Chrome's V8 JavaScript engine.
+ Platform.sh supports deploying Node.js applications quickly and easily.
+
+## Supported versions
+
+* 0.12.7 (default)
```
Comment (ralt) [inline reference/nodejs.md]: 4.2.1 will be the default version

File: reference/nodejs.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Configuration
+
+To use Platform.sh and Node.js together, configure the ``.platform.app.yaml``
+file with a few key settings, as described here (a complete example is included
+at the end).
+
+1. Specify the `type` as ``nodejs``
```
Comment (ralt) [inline reference/nodejs.md]: The version should be in the type, e.g. `nodejs:4.2` or `nodejs:0.12`

File: reference/nodejs.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+2. Specify your dependencies under the `nodejs` key, like this:
```
Comment (ralt) [inline reference/nodejs.md]: These are the build time dependencies, it should be made explicit

File: reference/nodejs.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+   ```yaml
+   dependencies:
+     nodejs:
+       'pm2': "*"
+       'react': "^0.13.0"
+       'url-loader': "~0.5.5"
+   ```
+   If you have a `package.json` file at the root of your project you can
```
Comment (ralt) [inline reference/nodejs.md]: No, `npm install` should be run in the build hook, which will install the dependencies specified in the `package.json`

File: reference/nodejs.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    'pm2': '*'
+    'apicache': "0.0.12"
+    'react': "^0.13.0"
+    'url-loader': "~0.5.5"
+
+web:
+  commands:
+    start: "PM2_HOME=/app/public pm2 start start.js --no-daemon"
+
+  locations:
```
Comment (ralt) [inline reference/nodejs.md]: This section isn't explained?

File: reference/nodejs.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+Finally, make sure your Node.js application is configured to listen over the
+environment variable ``PORT``, and you're done. Here's an example:
+
+```javascript
+var http = require('http');
+   
+http.createServer(function (req, res) {
+  res.writeHead(200, {'Content-Type': 'text/plain'});
+  res.end('Hello World\n');
+}).listen(process.env.PORT, "127.0.0.1");
```
Comment (ralt) [inline reference/nodejs.md]: Why listening on 127.0.0.1?

**Comment by sruggier** [comment]
Sorry, my mistake, I didn't mean to close this.

**Pull request closed without merging**